editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java

```diff
@@ -0,0 +1,357 @@
+      try {
+        // get the list of results from the future
+        List<IdentifyLayerResult> identifyLayerResults = identifyLayerResultsFuture.get();
+        // if one or more results have been identified
+        if (!identifyLayerResults.isEmpty()) {
+          // retrieve the first result
+          IdentifyLayerResult layerResult = identifyLayerResults.get(0);
+          // check that the result is a feature layer, thereby excluding annotation layers
+          if (layerResult.getLayerContent() instanceof FeatureLayer) {
+            // get a reference to the identified feature
+            selectedFeature = (Feature) layerResult.getElements().get(0);
+            // check the geometry and select the feature
+            selectFeature(layerResult);
+          }
+        }
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
+    });
+  }
+
+  /**
+   * Checks if the identified feature is a straight polyline or a point, and select the feature.
+   * For a point feature, show a dialog to edit attributes. Future clicks will call move functions.
+   *
+   * @param layerResult identify layer result
+   */
+  private void selectFeature(IdentifyLayerResult layerResult) {
+
+    try {
+      // if the selected feature is a polyline
+      if (selectedFeature.getGeometry().getGeometryType() == GeometryType.POLYLINE) {
+        // create a polyline builder from the selected feature
+        PolylineBuilder polylineBuilder = new PolylineBuilder((Polyline) selectedFeature.getGeometry());
+        // get a list of parts of the selected polyline
+        List<Part> parts = polylineBuilder.getParts();
+        parts.forEach(part -> {
+          // if the selected feature is a polyline with any part containing more than one segment
+          // (i.e. a curve)
+          if (part.getPointCount() > 2) {
+            selectedFeature = null;
+            // show message reminding user to select straight (single segment) polylines only
+            new Alert(Alert.AlertType.WARNING, "Select straight (single segment) polylines only.").show();
+            // return early, effectively disallowing selection of multi segmented polylines
+            return;
+          } else {
+            // select the polyline feature
+            ((FeatureLayer) layerResult.getLayerContent()).selectFeature(selectedFeature);
+            selectedFeatureIsPolyline = true;
+          }
+        });
+      }
+      // if the selected feature is a point, select the feature
+      else if (selectedFeature.getGeometry().getGeometryType() == GeometryType.POINT) {
+        ((FeatureLayer) layerResult.getLayerContent()).selectFeature(selectedFeature);
+        // open a dialog to edit the feature's attributes
+        showEditableAttributes(selectedFeature);
+      } else {
+        new Alert(Alert.AlertType.WARNING, "Feature of unexpected geometry type selected.").show();
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
```

Comment:
It doesn't look like there's any need for this try/catch block anymore, since all the asynchronous actions are now being done in other methods? In which case we can just remove it